c/driver/sqlite/statement_reader.c

```diff
@@ -120,6 +120,12 @@ static AdbcStatusCode ArrowTimestampToIsoString(int64_t value, enum ArrowTimeUni
       break;
   }
 
+  rem = value % scale;
```

Comment:
Fixes a bug where negative values where only changing the fractional part and not offseting the date, i.e. you'd get `1970-01-01T00:00:00.958` with ms precision instead of the expected `1969-12-31T23:59:59.958`